lib/output.js

```diff
@@ -887,6 +904,21 @@ function OutputStream(options) {
         self._do_print(output, "throw");
     });
 
+    /* -----[ yield ]----- */
+
+    DEFPRINT(AST_Yield, function(self, output){
+        var star = self.is_star ? "*" : "";
+        output.print("yield" + star);
+        if (self.expression) {
+            output.space();
+            self.expression.print(output);
+        } else if (self.is_star === true) {
```

Comment:
Good point. The expression is mandatory for a `yield*`.
```
     OPT(AST_Yield, function(self, compressor){
-        if (self.expresssion instance of AST_Undefined) {
+        if (self.expresssion instance of AST_Undefined && !self.is_star) {
             self.expresssion = null;
         }
         return self;
    });
```
FWIW, I wasn't aware of `yield*` before this PR and don't know how it works. ;-)